src/fmu/tools/rms/zone_mapping.py

```diff
@@ -0,0 +1,242 @@
+
+        assert len(zone_names_from_grid) == len(zonation)
+
+        # same index in zonation and in zone_names corresponds to the same zone
+        for zone_index in zonation:
+            zone_name = zone_names_from_grid[zone_index]
+            self.grid_zone_dict[zone_index]["name_from_grid"] = zone_name
+
+            layer_ranges = zonation[zone_index]
+            assert (
+                len(layer_ranges) == 1
+            )  # No repeated layer numbering for simbox indexer
+            layer_range = layer_ranges[0]
+            start = layer_range[0]
+            end = layer_range[-1]
+            number_of_layers = end + 1 - start
+            self.grid_zone_dict[zone_index]["nlayers"] = number_of_layers
+            self.grid_zone_dict[zone_index]["start_layer"] = start
+            self.grid_zone_dict[zone_index]["end_layer"] = end
+
+            zone_number = self.get_zone_number(zone_name)
+            self.grid_zone_dict[zone_index]["zone_number"] = zone_number
+            self.grid_zone_index_dict[zone_number] = zone_index
+
+    def validate_zone_number_for_grid(self, zone_number: int):
+        if zone_number not in self.grid_zone_index_dict:
+            if zone_number in self.zone_code_names:
+                raise ValueError(
+                    f"Zone name: {self.zone_code_names[zone_number]} "
+                    f"with zone code: {zone_number} in zone parameter "
+                    "does not exist in the grid."
+                )
+            raise ValueError(
+                f"Zone number: {zone_number} is not found in zone parameter."
+            )
+
+    def get_number_of_zones_in_grid(self):
+        return len(self.grid.simbox_indexer.zonation)
+
+    def get_zone_names_from_grid(self):
+        """
+        Zone names got from the grid object.
+        """
+        return self.grid.zone_names
+
+    def get_zone_names_from_param(self):
+        """
+        Zone names corresponding to the grid zones, but the names are
+        got from the zone parameter.
+        """
+        zone_name_dict = {}
+        nzones = self.get_number_of_zones_in_grid()
+        for zone_index in range(nzones):
+            zone_number = self.get_zone_number_for_zone_index(zone_index)
+            zone_name_dict[zone_number] = self.zone_code_names[zone_number]
+        return zone_name_dict
+
+    def get_zone_numbers(self):
+        return list(self.zone_code_names.keys())
+
+    def get_zone_numbers_in_grid(self):
+        nzones = self.get_number_of_zones_in_grid()
+        zone_numbers = []
+        for zone_index in range(nzones):
+            zone_numbers.append(self.grid_zone_dict[zone_index]["zone_number"])
+        return zone_numbers
+
+    def get_zone_number(self, zone_name: str):
+        # Unique zone name for each zone code
+        for code, name in self.zone_code_names.items():
+            if zone_name == name:
+                return code
+        raise ValueError(f"Unknown zone name: {zone_name}")
```

Comment:
In general (applies to this and almost all functions): If using type annotations, also indicate return type
, e.g.

`def get_zone_number(self, zone_name: str) -> int:`